### PR TITLE
rdma-core: update to 44.0.

### PR DESCRIPTION
**Pull request by kruceter**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-musl
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - armv6l-musl (cross)

### CODE REVIEW

**Comment by classabbyamp** [comment]
any common/shlibs entries from the removed package should be deleted

File: srcpkgs/rdma-core/template [hunk 1]
```diff
@@ -24,21 +26,18 @@ conf_files="
...
-	   ${DESTDIR}/usr/lib/udev/rules.d
-
-	# The cmake is bad
-	mv ${DESTDIR}/usr/etc/* ${DESTDIR}/etc/
+post_patch() {
+	# install manpages not depending on pandoc and rst2man
+	sed -i buildlib/rdma_man.cmake \
+		-e 's|"\${I}" MATCHES "\\\\\.md\$"|& AND PANDOC_FOUND|' \
+		-e 's|"\${I}" MATCHES "\\\\\.in\\\\\.rst\$"|& AND RST2MAN_FOUND|'
+}
```
Comment (ahesford) [inline srcpkgs/rdma-core/template:34]: Don't abuse `post_patch` for dynamic patching that doesn't need to be dynamic, especially not with `sed` which will happily and silently do nothing when your patterns cease to apply. Make a proper patch.

File: srcpkgs/rdma-core/template [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
 short_desc="RDMA core userspace libraries and daemons"
 maintainer="Rich G <rich@richgannon.net>"
-license="GPL-2.0-or-later, BSD-2-Clause-FreeBSD"
+license="GPL-2.0-or-later, BSD-2-Clause"
 homepage="https://github.com/linux-rdma/rdma-core"
 distfiles="https://github.com/linux-rdma/rdma-core/releases/download/v${version}/rdma-core-${version}.tar.gz"
-checksum=0afb7aa2654ea914c3b5d12e6e8f2fc3f325b0a52da1febf4f66788bf0b2bb14
+checksum=05ae494efb2b6c86f9586ce75a9c959995cb4b122fcb36ed4a32f96c9650017d
+build_options="docs"
+desc_option_docs="Enable documentation"
```
Comment (ahesford) [inline srcpkgs/rdma-core/template:17]: Enable this by default for platforms that provide `pandoc`.

Comment (kruceter) [reply srcpkgs/rdma-core/template:17]: Done. Thank you!

**Comment by kruceter** [comment]
> any common/shlibs entries from the removed package should be deleted

`common/shlibs` does not have any entries for libibmad.

**Review by ahesford — CHANGES_REQUESTED** [review]
Adding `libibmad*` to `removed-packages` is not the right approach here. Instead, add transitional dummies for `libibmad` and `libibmad-devel` as subpackages of `rdma-core`. `libibmad` should pull in `rdma-core` and `libibmad-devel` should pull in `rdma-core-devel`. This is important for two reasons:

1. `libibmad` has been subsumed by `rdma-core`, not really "removed". This is an important distinction for people who might attempt partial updates.
2. Just dropping the template will still leave the old `libibmad` in repositories, which will stage the repos the next time `rdma-core` bumps the soname of `libibumad.so.3`. That would require manual cleanup, usually at the least opportune time.

Because the version number of `rdma-core` is greater than that of the existing `libibumad`, the transitional package can take on the `rdma-core` version number and therefore does not need a separate template.

**Comment by kruceter** [comment]
> 1. `libibmad` has been subsumed by `rdma-core`, not really "removed". This is an important distinction for people who might attempt partial updates.

I have been under the impression the unused package (no reverse dependencies) can be safely removed, but since some people might *need* `libibmad`, I see no harm to try and symlink `rdma-core`.

> 2\. Just dropping the template will still leave the old `libibmad` in repositories, which will stage the repos the next time `rdma-core` bumps the soname of `libibumad.so.3`. That would require manual cleanup, usually at the least opportune time.

libibmad and libibumad are two different libraries, Andrew.

Even if `libibumad`'s soname gets changed at some point, libibmad plays no role here.

**Pull request closed without merging**